.github/actions/create-project-issue-v1/src/run.test.ts

```diff
@@ -0,0 +1,308 @@
+import 'mocha'
+import { assert } from 'chai'
+import * as exec from '@actions/exec'
+import sinon from 'sinon'
+import type { Core, GitHub } from './types'
+import run from './run'
+import { MOCK_PROJECT_BOARD_ID } from './getProjectBoardID.test'
+import { MOCK_ISSUE_ADDED } from './addIssueToBoard.test'
+import { MOCK_FIELD_LIST } from './getProjectFieldList.test'
+import { MOCK_ISSUE_MOVED } from './moveIssueToColumn.test'
+
+describe('run', () => {
+  let setFailed: sinon.SinonSpy
+  let setOutput: sinon.SinonStub
+  let info: sinon.SinonSpy
+  let getInput: sinon.SinonStub
+  let getExecOutput: sinon.SinonStub
+  const github = {
+    context: {
+      repo: {
+        owner: 'owner',
+        repo: 'repo'
+      }
+    },
+    getOctokit: () => ({
+      rest: {
+        issues: {
+          create: sinon.stub().resolves({
+            data: {
+              html_url: 'https://deque.bizzy.com',
+              node_id: 'generated-node-id'
+            }
+          })
+        }
+      }
+    })
+  }
+
+  beforeEach(() => {
+    setFailed = sinon.spy()
+    setOutput = sinon.stub()
+    info = sinon.spy()
+    getInput = sinon.stub()
+    getExecOutput = sinon.stub(exec, 'getExecOutput')
+  })
+
+  afterEach(sinon.restore)
+
+  const createAndMoveIssueStub = async () => {
+    //`getProjectBoardID` stub
+    getExecOutput.onFirstCall().resolves({
```

Comment:
This relies on the order of events which isn't robust. Instead you can use `stub.withArgs` and [sinon matchers](https://sinonjs.org/releases/latest/matchers/) to match the specific calls to their responses. By specifying more detailed matches you can even give different responses based on input.

```suggestion
    getExecOutput.withArgs(sinon.match('gh project view')).resolves({
```